src/ui/public/directives/kbn_top_nav.js

```diff
@@ -26,6 +26,8 @@ module.directive('kbnTopNav', function (Private) {
       label: _.capitalize(opt.key),
       hasFunction: !!opt.run,
       description: ('Toggle ' + opt.key),
+      noButton: !!opt.noButton,
+      openByDefault: !!opt.openByDefault ,
```

Comment:
I'm not a fan of this, why can't you just call `kbnTopNav.open()`? It seems this shouldn't really be this directives responsibility.